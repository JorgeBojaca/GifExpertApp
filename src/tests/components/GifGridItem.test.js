import { shallow } from "enzyme";
import GifGridItem from "../../components/GifGridItem";

describe("Prueba para componente GifGridItem", () => {
  const title = "Título de prueba";
  const url = "http://localhost/imagen.jpg";
  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  test("Debe mostrar el componente correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe tener un párrafo con el texto de la prop title ", () => {
    const p = wrapper.find("p");
    expect(p.text().trim()).toBe(title);
  });

  test("Debe tener una img con las propiedades src=$url y alt=$title ", () => {
    const img = wrapper.find("img");
    expect(img.prop("src")).toBe(url);
    expect(img.prop("alt")).toBe(title);
  });

  test("Debe tener la clase 'animated__fadeIn'", () => {
    const div = wrapper.find("div");
    const classText = div.prop("className").split(" ");
    const hasClass = classText.includes("animated__fadeIn");
    expect(hasClass).toBe(true);
  });
});
