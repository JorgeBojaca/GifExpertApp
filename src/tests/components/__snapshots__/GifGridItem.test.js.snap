// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba para componente GifGridItem Debe mostrar el componente correctamente 1`] = `
<div
  className="gif-grid-item animated__fadeIn"
>
  <img
    alt="Título de prueba"
    src="http://localhost/imagen.jpg"
  />
  <p>
    Título de prueba
  </p>
</div>
`;
